### DISCUSSION TITLE
Calibrate touch flag

### DISCUSSION THREAD

**Discussion by DougK71** [rzeldent/esp32-smartdisplay#94] [Ideas]
Since good calibration is necessary a calibrate #define CALIBRATE if present in the lvgl.h  if uncommitted would 
cause a calibration or recalibration to occur on powerup and store the touch  values
I'm unsure whether they go into eeprom on the touch chip or on the cpu. So far it appears calibration from the demo  is stored and is read in for landscape mode from a prior portrait mode calibration  **smartdisplay_compute_touch_calibration(screen[3], touch[3]);** looks like it would do the calibration but I am having trouble calling it and understanding if the calibration indeed goes to EEPROM the need to expose the values via screen[3] and touch[3]

**Comment by DougK71** [comment]
I am impressed with the handling of the board display interface and the almost seamless of designing in Squareline Studio and pasting in the ui folder under scr.

**Reply by DougK71** [reply]
This request is not needed as there is in the  esp32-8048S070C.jason in Users>Yourname>.platformio>platforms>espressif32>boards are specified in addition to PIN assigns 
 "-DLCD_WIDTH=800",
 "-DLCD_HEIGHT=480",
"-DTOUCH_SWAP_X=false",  original was true and messed up landscape
 "-DTOUCH_SWAP_Y=false", original was true and messed up landscape 

This provides the data about the mapping between the touch control  domain to the screen pixel domain at the level of the board effectively melding the LCD orientation  and MCU and Touch driver the description of the board using platformio

**Reply by rzeldent** [reply]
Hi Doug,

Watch out for putting the board definitions in the Users>Yourname>.platformio>platforms>espressif32>boards directory. They should be in the boards directory in the main root of the project. Normally this is done as a submodule so it keeps up with the (fixes in) definitions (json)...

**Comment by rzeldent** [comment]
> Since good calibration is necessary a calibrate #define CALIBRATE if present in the lvgl.h if uncommitted would cause a calibration or recalibration to occur on powerup and store the touch values I'm unsure whether they go into eeprom on the touch chip or on the cpu. So far it appears calibration from the demo is stored and is read in for landscape mode from a prior portrait mode calibration **smartdisplay_compute_touch_calibration(screen[3], touch[3]);** looks like it would do the calibration but I am having trouble calling it and understanding if the calibration indeed goes to EEPROM the need to expose the values via screen[3] and touch[3]

The calibration does not go into the eprom but managed by the user. This is, if enabled, applied to the input of the touch before reported to LVGL. There is also no calibration screen build in but this on the road map. For more information how to calculate see: https://www.analog.com/media/en/technical-documentation/application-notes/an-1021.pdf
